docs/manual/guide/services/HeaderFilters.md

```diff
@@ -0,0 +1,30 @@
+# Header Filters
+
+In Lagom you may add `HeaderFilter`s to your `ServiceDescriptor`. In a `HeaderFilter` you will usually handle protocol negotiation or authentication. 
+
+A single `HeaderFilter` implementation may transform a request leaving a client or entering a server and a response leaving a server and entering a client.
+
+@[user-agent-auth-filter](code/docs/services/UserAgentHeaderFilter.java)
+
+This [`UserAgentHeaderFilter`](api/index.html?com/lightbend/lagom/javadsl/api/security/UserAgentHeaderFilter.html) is the default `HeaderFilter` any Lagom service will use if none is specified. It uses a `ServicePrincipal` which identifies the client with the service name. This way a server may identify a caller client by the `User-Agent`. 
+
+In `UserAgentHeaderFilter` the code at `transformClientRequest` will be invoked when preparing a client invocation to add a `User-Agent` header if a [ServicePrincipal](#ServicePrincipal) was specified on the request. On the server end `transformServerRequest` will be used to read the `User-Agent` header and set that value as the request's [Principal](https://docs.oracle.com/javase/8/docs/api/java/security/Principal.html).
+
+Keep in mind that a HeaderFilter should not be used to alter the request/response processing, it is just a tool to adapt two ends. 
```

Comment:
Perhaps:

Keep in mind that a header filter should only be used to deal with cross cutting protocol concerns, and nothing more. For example, you may have a header filter that describes how the current authenticated user is communicated over HTTP (by adding a user header, for example). Cross cutting domain concerns, such as authentication and validation, should not be handled in a header filter, rather they should be handled using [[service call composition|link-to-service-call#composition]].